machine/v1beta1/types_awsprovider.go

```diff
@@ -95,6 +95,17 @@ type AWSMachineProviderConfig struct {
 	// The field size should be greater than 0 and the field input must start with cr-***
 	// +optional
 	CapacityReservationID string `json:"capacityReservationId"`
+	// marketType specifies the type of market for the EC2 instance.
+	// Valid values are OnDemand, Spot, and CapacityBlock.
+	// Defaults to OnDemand.
+	// When set to OnDemand the instance runs as a standard OnDemand instance.
+	// When set to Spot the instance runs as a Spot instance. When SpotMarketOptions is provided, the marketType defaults to "Spot".
+	// When set to CapacityBlock the instance utilizes pre-purchased compute capacity (capacity blocks) with AWS Capacity Reservations.
+	// If this value is selected, CapacityReservationID must be specified to identify the target reservation.
+	// If marketType is not specified and spotMarketOptions is provided, the marketType defaults to "Spot".
+	// +kubebuilder:validation:XValidation:rule="has(self.marketType) && self.marketType == 'CapacityBlock' ?  has(self.capacityReservationId) : !has(self.capacityReservationId)",message="capacityReservationId is required when marketType is CapacityBlock, and forbidden otherwise"
```

Comment:
This might need to go on the type as a whole, but if I understood @JoelSpeed's comment about these API types not having a generated CRD schema this validation won't actually do anything?

Would you be able to link me to the changes that are going to be made to the existing validating webhook that implements this validation?

Reply:
not yet implemented will implement after this merge. 
You can find [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/cc12c529c059bc0fd3b85d30da13a26642fbba3a/api/v1beta2/awsmachine_webhook.go#L365-L377) is the upstream validation, same will be implemented here.